src/sampler/reserve.jl

```diff
@@ -0,0 +1,103 @@
+import Distributions: length, eltype, _rand!
+
+abstract type AbstractReserveSampler end
+
+"""
+    E2ELRReserveScaler{U,F}
+
+Samples reserve requirements following the procedure below:
+
+1. Sample a minimum reserve requirement `MRR` from a uniform distribution `U(lb, ub)`.
+2. Compute the upper bound of reserve requirements for each generator as `rmax = α * (pmax - pmin)`.
+3. Fix the lower bound of reserve requirement per generator to zero.
+
+The parameter `α` is a scaling factor that determines the total reserve requirement
+    as a multiple of the maximum power range of all generators.
+
+"""
+struct E2ELRReserveScaler <: AbstractReserveSampler
+    mrr_dist::Uniform
```

Comment:
`Uniform` is an abstract type (see its definition [here](https://github.com/JuliaStats/Distributions.jl/blob/6af1e2f2f2448006e308f4ab95ca0f66037563e6/src/univariate/continuous/uniform.jl#L26)), you may want to write `::Uniform{Float64}` instead.

Reply:
Any concerns here about supporting Float128? I guess the reference data is only Float64 anyway.

Reply:
I believe we can live with not supporting `Float128` here.
We've been using `ScaledLogNormal` which is `Float64` and so far, we're good.

If we do want to support arbitrary precision in the sampler, I see 2 possibilities:
* we decide to be OK with a little bit of type instability
* we add an extra type parameter to represent the data type

The former is not a huge loss. Code will run _slightly_ slower, but not noticeable compared to, e.g., solving the OPFs.
The latter increases the maintenance burden on us.